tests/compiler/std/typedarray.optimized.wat

```diff
@@ -3380,30 +3380,23 @@
   (local $2 i32)
   (local $3 f64)
   (local $4 f64)
-  block $~lib/typedarray/SORT<~lib/typedarray/Float64Array,f64>|inlined.0
+  local.get $0
+  call $~lib/typedarray/Int64Array#get:length
+  local.tee $2
+  i32.const 1
+  i32.gt_s
+  if
    local.get $0
-   call $~lib/rt/pure/__retain
-   local.tee $1
-   call $~lib/typedarray/Int64Array#get:length
-   local.tee $2
-   i32.const 1
-   i32.le_s
-   if
-    local.get $1
-    call $~lib/rt/pure/__release
-    br $~lib/typedarray/SORT<~lib/typedarray/Float64Array,f64>|inlined.0
-   end
-   local.get $1
```

Comment:
@dcodeIO It seems this PR still has improvements after rebased from #1043, so I leave it opened